ufs2arco/regrid_mom6.py

```diff
@@ -0,0 +1,302 @@
+        latitudes, _ = gaussian_latitudes(nlat // 2)
+        longitudes = np.linspace(0, 360, nlon, endpoint=False)
+        return latitudes, longitudes
+
+    @staticmethod
+    def compute_latlon_grid(nlat, nlon) -> (np.array, np.array):
+        """Compute regular latlong grid coordinates"""
+        latitudes = np.linspace(-90, 90, nlat, endpoint=False)
+        longitudes = np.linspace(0, 360, nlon, endpoint=False)
+        return latitudes, longitudes
+
+    @staticmethod
+    def read_grid(file, lats_name="grid_yt", lons_name="grid_xt") -> (np.array, np.array):
+        """Read grid latitudes and longitudes from a netcdf file"""
+        ds = xr.open_dataset(file)
+        latitudes = ds[lats_name].values
+        longitudes = ds[lons_name].values
+        ds.close()
+        return latitudes, longitudes
+
+    def create_regridder(self, ds_in: xr.Dataset) -> None:
+        """
+        Create regridder: computes weights and creates three xesmf.Regridder instances.
+        :param ds_in: MOM6 xarray dataset needed for creating the regridder.
+        """
+        self.rotation_file = self.config["rotation_file"]
+
+        # open input dataset
+        ds_in_t = ds_in.rename({"xh": "lon", "yh": "lat"})
+        ds_in_u = ds_in.rename({"xq": "lon", "yh": "lat"})
+        ds_in_v = ds_in.rename({"xh": "lon", "yq": "lat"})
+
+        # open rotation dataset
+        ds_rot = xr.open_dataset(self.rotation_file)
+        ds_rot = ds_rot[["cos_rot", "sin_rot"]]
+        self.ds_rot = ds_rot.rename({"xh": "lon", "yh": "lat"})
+
+        # create output dataset
+        lons, lats = np.meshgrid(self.lons1d_out, self.lats1d_out)
+        grid_out = xr.Dataset()
+        grid_out["lon"] = xr.DataArray(lons, dims=["nx", "ny"])
+        grid_out["lat"] = xr.DataArray(lats, dims=["nx", "ny"])
+
+        # get nlon/nlat for input/output datsets
+        nlon_i = ds_in.sizes["yh"]
+        nlat_i = ds_in.sizes["xh"]
+        nlon_o = len(self.lons1d_out)
+        nlat_o = len(self.lats1d_out)
+        self.ires = f"{nlon_i}x{nlat_i}"
+        self.ores = f"{nlon_o}x{nlat_o}"
+
+        # paths to interpolation weights files
+        if "weights_file_t2t" in self.config.keys():
+            weights_file_t2t = self.config["weights_file_t2t"]
+            weights_file_u2t = self.config["weights_file_u2t"]
+            weights_file_v2t = self.config["weights_file_v2t"]
+        if weights_file_t2t is None:
+            weights_file_t2t = f"weights-{self.ires}.Ct.{self.ores}.Ct.{self.interp_method}.nc"
+            weights_file_u2t = f"weights-{self.ires}.Cu.{self.ires}.Ct.{self.interp_method}.nc"
+            weights_file_v2t = f"weights-{self.ires}.Cv.{self.ires}.Ct.{self.interp_method}.nc"
+
+        # create regridding instances
+        reuse = os.path.exists(weights_file_t2t)
+        self.rg_tt = xe.Regridder(
+            ds_in_t,
+            grid_out,
+            self.interp_method,
+            periodic=True,

```

Comment:
We won't always have periodic data, and I suggest we deal with this now before we forget about it. Could we move this to the config yaml as an option? Or we can warn the user, but I think it's as much effort to set up the warning as it is to make the option in the yaml.

Reply:
Added periodic to the yaml file.